### PR TITLE
(feat): Functional test suite

### PR DESCRIPTION
**Pull request by dimtruck**
UPDATE: due to the protocol being in flux, waiting on adding the metric validation 'til next PR.  Test stub is added here; however, so addition will be minimal.  This PR is now ready for review.

* added endpoint and serve integration tests
* added test configurations and certs
* updated test utils to include functions to execute CLI rackspace poller commands

### CODE REVIEW

File: integrationcli/endpoint_test.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+	"github.com/racker/rackspace-monitoring-poller/config"
+	"github.com/racker/rackspace-monitoring-poller/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStartEndpoint(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	/*
```
Comment (rphillips) [inline integrationcli/endpoint_test.go]: we should remove the dead code if we aren't using it.

File: integrationcli/testdata/server-certs/bundle.pem [hunk 2]
```diff
@@ -0,0 +1,40 @@
+-----BEGIN CERTIFICATE-----
```
Comment (rphillips) [inline integrationcli/testdata/server-certs/bundle.pem]: would be nice to reuse the certificates we have... or use a library to generate dummy certificates

File: integrationcli/endpoint_test.go (hunk 1 above)
Comment (dimtruck) [inline integrationcli/endpoint_test.go]: Sorry, definitely

**Comment by dimtruck** [comment]
@rphillips @itzg  Thinking of adding tests to the utils/test.go file, what do you guys think?  Useful?

**Comment by dimtruck** [comment]
It would require to add some interfaces and extract direct calls to exec package, which is fine.

**Comment by itzg** [comment]
@dimtruck do you mean tests **of** test/utils.go? A bit inception, but sounds good to me :)

**Comment by rphillips** [comment]
+1

**Pull request merged**